lib/puppet-strings/hiera/data.rb

```diff
@@ -64,13 +78,40 @@ def to_s
 
     private
 
+    def value_to_s(value)
+      if value.is_a?(Array) && value.empty?
+        return '[]'
+      elsif value.is_a?(Hash) && value.empty?
+        return '{}'
+      elsif value.nil?
+        return 'undef'
+      else
+        return value.to_s
+      end
+    end
+
+    # @return [Hash[String, Hash[String, Any]]]
+    #   variable -> filename: value
+    def filter_mappings(class_name, mappings)
+      result = {}
+      mappings.each do |key, value|
+        mapped_class_name, _, variable = key.rpartition('::')
+        if mapped_class_name == class_name
+          result[variable] = value
+        end
+      end
+      result
+    end
+
+    # TODO: this should be a class method not an instance method
     def load_config
       return unless File.exist?(config_path)
 
       config = YAML.load(File.read(config_path))
 
       unless config['version'] == 5
-        raise "Unsupported version '#{config['version']}'"
+        log.warn("Unsupported version '#{config['version']}'")
+        return
```

Comment:
i think this achieves "Graceful fallback if legacy Hiera data is used" although see the todo above, currently we load the config for every instance